main.go

```diff
@@ -251,3 +328,21 @@ func rawSetuid(uid int) error {
 
 	return nil
 }
+
+func (handler *BuildHandler) saveNewBuild(repoName string) BuildInfo {
+	info := BuildInfo{
+		repository: repoName,
+		startedAt:  time.Now(),
+		status:     "in progress",
+
+		pointer: handler.currentBuild,
```

Comment:
No need of such complicated implementation. Just initialize `BuildInfo{}` struct before `runBuild` call (remove it from `saveNewBuild`), therefore, there are not need for `pointer` and `BuildInfo.Save()`.